src/move_mode.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+#ifndef MOVE_MODE_H
+#define MOVE_MODE_H
+
+#include <cstddef>
+#include <cstdint>
+#include <algorithm>
+#include <vector>
+#include <memory>
+#include <string>
+
+#include "type_id.h"
+#include "string_id.h"
+#include "translations.h"
+#include "color.h"
+
+class JsonObject;
+
+struct move_mode {
+    public:
+        void load( const JsonObject &jo, const std::string &src );
+        void finalize();
+        void check() const;
+
+    public:
+        // Used by generic_factory
+        move_mode_str_id id;
+        bool was_loaded = false;
+
+    public:
```

Comment:
i suggest you make every single one of these private.

Reply:
Should these members instead be accessed through functions?

Reply:
yes, absolutely. set_ and get_ and maybe functions with context names. was_loaded is an exception, it needs to be public because of generic_factory. i can't recall off hand if id is the same